cache/metrics.go

```diff
@@ -0,0 +1,85 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const (
+	instrumentationName      = "github.com/moby/buildkit/cache"
+	metricCacheRecords       = "cache.records.count"
```

Comment:
I was wondering if we should have a `cache.records.size` to collect the cache size or maybe split between `cache.records.shared.size`, `cache.records.private.size`, `cache.records.reclaimable.size`?

Reply:
I'm not really sure to be honest. I do think that will likely take a bit more effort though to implement so I left it out of this initial version. The reason is because I wasn't quite sure how to determine the disk usage in an efficient way taking into account mutable and immutable. I figured immutable entries didn't need to be continuously updated while mutable ones would have to be rechecked. I also didn't want to hold a lock on the disk usage or perform a potentially expensive computation to count the size.

I do think it's likely worth making a new issue for cache sizes and adding some metrics to it. I'd say all of the above are good metrics.

Reply:
> to hold a lock on the disk usage or perform a potentially expensive computation to count the size.

Oh right that's a very good point, disk usage is indeed a resource intensive call.